pcdsdevices/tests/test_interface.py

```diff
@@ -191,6 +201,55 @@ def test_presets_type(presets, fast_motor):
         fast_motor.presets.add_user(234234, 'cats')
 
 
+def test_presets_desync(presets, fast_motor: FastMotor):
+    assert not fast_motor.presets.sync_needed
+
+    fast_motor.mv(4, wait=True)
+    fast_motor.presets.add_hutch('four', comment='four!')
+
+    assert not fast_motor.presets.sync_needed
+
+    # modify preset from other object with the same, to force collision
+    fast_motor2 = FastMotor(name='sim_fast')
+    fast_motor2.mv(5, wait=True)
+    fast_motor2.presets.positions.four.update_pos()
+
+    # in-memory python objects have different preset positions
+    assert fast_motor.presets.positions.four.pos == 4
+    assert fast_motor2.presets.positions.four.pos == 5
+
+    # but point to the same file
+    assert fast_motor.presets._path("hutch") == fast_motor2.presets._path("hutch")
+
+    # original object needs a sync, but second does not
+    assert fast_motor.presets.sync_needed
+    assert not fast_motor2.presets.sync_needed
+
+
+def test_presets_tab_init(fast_motor: FastMotor, deferred_fast_motor_presets):
+    # deferred_fast_motor_preset must come last,
+    # to clear cache after motor is created (and sync-ed at init)
+    assert fast_motor.presets.sync_needed
+    fast_motor.__dir__()  # mimic tab completion request
```

Comment:
Nitpick: is `dir(fast_motor)` equivalent?

Reply:
I believe so, do we prefer that to the dunder-access?  I used `__dir__` just to match the code usage.